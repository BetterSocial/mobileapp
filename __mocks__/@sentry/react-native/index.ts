export const addBreadcrumb = jest.fn();
