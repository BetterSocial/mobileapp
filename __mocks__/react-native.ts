import * as RN from 'react-native';

RN.Animated.timing = () => ({
    start: () => jest.fn()
})

module.exports = RN