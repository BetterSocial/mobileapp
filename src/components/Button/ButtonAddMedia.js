import * as React from 'react';
import {StyleSheet, Text, TouchableOpacity} from 'react-native';

import {colors} from '../../utils/colors';

const ButtonAddMedia = ({onPress, style, label, labelStyle}) => {
  return (
    <TouchableOpacity style={[styles.container, style]} onPress={onPress}>
      <Text style={labelStyle}>{label}</Text>
    </TouchableOpacity>
  );
};

export default ButtonAddMedia;

const styles = StyleSheet.create({
  container: {
    paddingVertical: 32,
    alignItems: 'center',
    borderWidth: 2,
    borderColor: colors.alto,
    borderRadius: 8,
    borderStyle: 'dashed',
  },
});
