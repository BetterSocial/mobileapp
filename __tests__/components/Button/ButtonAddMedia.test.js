import * as React from 'react';
import { render, cleanup } from '@testing-library/react-native';
import ButtonAddMedia from '../../../src/components/Button/ButtonAddMedia';

const mockNavigate = jest.fn()

jest.mock('@react-navigation/core', () => ({
  ...jest.requireActual('@react-navigation/core'),
  useNavigation: () => ({
    goBack: jest.fn(), navigate: mockNavigate 
  }),
}));


describe('ButtonAddMedia should run correctly', () => {
// onPress, style, label, labelStyle}
    afterEach(cleanup)
    it('should match snapshot', () => {
        const handleOnpress = jest.fn()
        const {toJSON, getAllByText} = render(<ButtonAddMedia label={'labelku'} onPress={handleOnpress} style={{backgroundColor: 'red'}} labelStyle={{color: 'blue'}} />)
        expect(toJSON).toMatchSnapshot()
        expect(getAllByText('labelku')).toHaveLength(1)
    })
})