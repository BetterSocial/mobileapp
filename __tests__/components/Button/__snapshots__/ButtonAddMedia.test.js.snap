// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonAddMedia should run correctly should match snapshot 1`] = `[Function]`;
